File: sitpass-bek/src/main/java/com/example/sitpassbek/service/ExerciseService.java
package com.example.sitpassbek.service;

import com.example.sitpassbek.dto.exercise.CreateExerciseDTO;
import com.example.sitpassbek.dto.exercise.ExerciseDTO;
import com.example.sitpassbek.dto.exercise.PastVisitsRequestDTO;

import java.util.List;

public interface ExerciseService {

    void deleteExercise(Long exerciseId);

    ExerciseDTO createExercise(CreateExerciseDTO dto);

    List<ExerciseDTO> getAllExercisesByUser(Long userId);

    Long getPastVisits(PastVisitsRequestDTO requestDto);

}

File: sitpass-front/src/app/services/exercise/exercise.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { CreateExercise, Exercise, PastVisitsRequest } from '../../types/exercise.type';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ExerciseService {

  private apiUrl = 'http://localhost:8080/exercises';

  constructor(private http: HttpClient) { }

  getExercisesByUser(userId: number): Observable<Exercise[]> {
    return this.http.get<Exercise[]>(`${this.apiUrl}/user/${userId}`);
  }

  createExercise(dto: CreateExercise): Observable<Exercise> {
    return this.http.post<Exercise>(this.apiUrl, dto);
  }

  deleteExercise(exerciseId: number) {
    return this.http.delete<void>(`${this.apiUrl}/${exerciseId}`);
  }

  getPastVisits(request: PastVisitsRequest): Observable<number> {
    return this.http.post<number>(`${this.apiUrl}/past-visits`, request);
  }

}
